src/components/tree-detail/tree-water-needs/contact-dialog.tsx

```diff
@@ -0,0 +1,245 @@
+/* eslint-disable max-lines */
+import React, { useCallback, useMemo, useState } from "react";
+import Markdown from "react-markdown";
+import { useAuthStore } from "../../../auth/auth-store";
+import { supabaseClient } from "../../../auth/supabase-client";
+import { useI18nStore } from "../../../i18n/i18n-store";
+import { useSelectedContactRecipientUsernameStore } from "../../../shared-stores/selected-contact-recipient-store";
+import { AlertDialog } from "../../alert-dialog/alert-dialog";
+import { WarningDialog } from "../../alert-dialog/warning-dialog";
+import { PrimaryButton } from "../../buttons/primary";
+import { TertiaryButton } from "../../buttons/tertiary";
+import { CheckIcon } from "../../icons/check-icon";
+import { CloseIcon } from "../../icons/close-icon";
+import { useTreeStore } from "../stores/tree-store";
+
+export const ContactDialog: React.FC = () => {
+	const { selectedContactRecipientUsername } =
+		useSelectedContactRecipientUsernameStore();
+
+	const recipientContactName = selectedContactRecipientUsername ?? "";
+
+	const selectedTreeId = useTreeStore().selectedTreeId;
+	const user = useAuthStore().session?.user;
+
+	const [message, setMessage] = useState("");
+	const [isContactRequestLoading, setIsContactRequestLoading] = useState(false);
+	const [error, setError] = useState("");
+
+	const i18n = useI18nStore().i18n();
+
+	const closeContactDialog = () => {
+		setMessage("");
+		setIsContactRequestLoading(false);
+		setError("");
+		(document.getElementById("contact-dialog") as HTMLDialogElement).close();
+	};
+
+	const containsLinks = useMemo(() => {
+		const urlRegex = /(https?:\/\/[^\s]+)/g;
```

Comment:
should we also prevent links like `www.google.com`, or `google.com`? Or is it too much?